pkg/tracegen/intake.go

```diff
@@ -23,24 +23,15 @@ import (
 	"context"
 	"errors"
 	"fmt"
-	"math"
 	"time"
 
 	"go.elastic.co/apm/v2"
+	"go.uber.org/zap"
 )
 
-type Config struct {
-	SampleRate float64
-	TraceID    apm.TraceID
-}
-
 // IndexIntakeV2Trace generate a trace including a transaction, a span and an error
-func IndexIntakeV2Trace(ctx context.Context, cfg Config, tracer *apm.Tracer) error {
-	if cfg.SampleRate < 0.0001 || cfg.SampleRate > 1.0 {
-		return errors.New("invalid sample rate provided. allowed value: 0.0001 <= sample-rate <= 1.0")
-	}
-	cfg.SampleRate = math.Round(cfg.SampleRate*10000) / 10000
-
+func IndexIntakeV2Trace(ctx context.Context, cfg Config, logger *zap.SugaredLogger) (apm.TraceContext, error) {
```

Comment:
I think it's better to use a dedicated config struct for this function. From a client's PoV, it'll provides clean separation and less confusion to what needs to be set in order for this to work.